workspaces/adventure-pack/goodies/typescript/Iterator.from/index.ts

```diff
@@ -0,0 +1,44 @@
+import { iteratorPrototype } from "../Iterator.prototype";
+import "../Iterator.prototype.toIterable";
+
+declare global {
+  interface IteratorConstructor {
+    from<T>(
+      object: Iterator<T> | Iterable<T> | { next(): IteratorResult<T> },
+    ): IterableIterator<T>;
+  }
+
+  const Iterator: IteratorConstructor;
+}
+
+(globalThis as Record<string, unknown>).Iterator ??= {};
+
+(globalThis as Record<string, unknown>).Iterator = {
+  from<T>(
+    object: Iterator<T> | Iterable<T> | { next(): IteratorResult<T> },
+  ): IterableIterator<T> {
```

Comment:
Since we're using the `toIterable` function so much, we can save it in a variable, and also give it a more specific type that will save us from some casts later on:

```ts
const toIterable = iteratorPrototype.toIterable as (
      this: Iterator<T>,
    ) => IterableIterator<T>;
```

Reply:
ack!